server/pkg/exporters/csv.go

```diff
@@ -98,19 +98,25 @@ func extractFirstPointField(itm *item.Item) ([]float64, error) {
 	return nil, noPointFieldError
 }
 
-func convertToCSV(data [][]string) (string, error) {
-	var sb strings.Builder
-	w := csv.NewWriter(&sb)
-	for _, row := range data {
-		if err := w.Write(row); err != nil {
-			return "", err
+func convertToCSV(pw *io.PipeWriter, data [][]string) error {
```

Comment:
no need to return an error here, instead the error must be checked in from the pipe in the upper layers